cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -748,38 +736,37 @@ func (p *BicepProvider) loadParameters(
 	return armParameters.Parameters, nil
 }
 
-// Creates the compiled template from the specified module path
-func (p *BicepProvider) createDeployment(ctx context.Context, modulePath string) (*Deployment, *azure.ArmTemplate, error) {
-	// Compile the bicep file into an ARM template we can create.
+func (p *BicepProvider) compileBicep(
+	ctx context.Context, modulePath string,
+) (azure.RawArmTemplate, azure.ArmTemplate, error) {
+
 	compiled, err := p.bicepCli.Build(ctx, modulePath)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to compile bicep template: %w", err)
+		return nil, azure.ArmTemplate{}, fmt.Errorf("failed to compile bicep template: %w", err)
```

Comment:
Not particularly - sort of a style, I guess.  I felt more strongly about forcing methods to take an `azure.ArmTemplate` instead of a `*azure.ArmTemplate` because I wanted it to be clear about these methods actually requiring a template (vs nil being a valid value).  At that point it felt reasonable to just return a value here, given that in general there's no guarantees about the return values when the err parameter is non nil.

Happy to move back if you'd like me to.

Reply:
Keeping what I have for now but happy to go back later.